### PR TITLE
Add output for failure cases

### PR DESCRIPTION
**Pull request by ostromart**

### CODE REVIEW

File: tests/upgrade/test_crossgrade.sh [hunk 1]
```diff
@@ -104,6 +104,7 @@ ISTIO_ROOT=${GOPATH}/src/istio.io/istio
 TMP_DIR=/tmp/istio_upgrade_test
```
Comment (hklai) [inline tests/upgrade/test_crossgrade.sh]: Prob better to put all theses in $ARTIFACTS_DIR so these logs get uploaded.
e.g. http://gcsweb.k8s.io/gcs/istio-prow/pr-logs/pull/istio-releases_pipeline/238/release-upgrade-tests-1.0.2/256/

File: tests/upgrade/test_crossgrade.sh [hunk 2]
```diff
@@ -129,13 +130,14 @@ withRetries() {
...
-      echo "RUNNING $*" ; "${@}" && break
+      echo "RUNNING $*" ; "${@}" > "${ERROR_TMP_LOG}" 2>&1 && break
       echo "Failed, sleeping ${sleep_sec} seconds and retrying..."
+      cat "${ERROR_TMP_LOG}"
       ((n++))
       sleep "${sleep_sec}"
     done
 
-    if (( n == max_retries )); then die "$* failed after retrying ${max_retries} times."; fi
+    if (( n == max_retries )); then cat "${ERROR_TMP_LOG}"; die "$* failed after retrying ${max_retries} times."; fi
```
Comment (hklai) [inline tests/upgrade/test_crossgrade.sh]: an then you don't need to cat the log file

File: tests/upgrade/test_crossgrade.sh (hunk 1 above)
Comment (ostromart) [reply tests/upgrade/test_crossgrade.sh]: these are just for temporary errors which are echoed to stdout in the test itself.

File: tests/upgrade/test_crossgrade.sh (hunk 2 above)
Comment (ostromart) [reply tests/upgrade/test_crossgrade.sh]: i could do it but i think outputting the error here will give more context about the error. wdyt?

**Comment by hklai** [comment]
Merging. This affects release qualification only.

**Pull request merged**